internal/downloadManager/download.go

```diff
@@ -30,116 +31,135 @@ func DownloadTools(config *configs.Config) error {
+		if errKubectl != nil {
+			log.Panicf("failed to call kubectlVersionCmd.Run(): %v", err)
+		}
+		wg.Done()
+	}()
+
+	go func() {
+		// todo: adopt latest helmVersion := "v3.9.0"
+		terraformVersion := config.TerraformVersion
+
+		terraformDownloadUrl := fmt.Sprintf(
+			"https://releases.hashicorp.com/terraform/%s/terraform_%s_%s_%s.zip",
+			terraformVersion,
+			terraformVersion,
+			config.LocalOs,
+			config.LocalArchitecture,
+		)
+		log.Printf("Downloading terraform from %s", terraformDownloadUrl)
+		terraformDownloadZipPath := fmt.Sprintf("%s/tools/terraform.zip", config.K1FolderPath)
+		err := downloadFile(terraformDownloadZipPath, terraformDownloadUrl)
+		if err != nil {
+			log.Println("error reading terraform file")
+			//return err
+		}
 
-	// todo: adopt latest helmVersion := "v3.9.0"
-	terraformVersion := config.TerraformVersion
-
-	terraformDownloadUrl := fmt.Sprintf(
-		"https://releases.hashicorp.com/terraform/%s/terraform_%s_%s_%s.zip",
-		terraformVersion,
-		terraformVersion,
-		config.LocalOs,
-		config.LocalArchitecture,
-	)
-	log.Printf("Downloading terraform from %s", terraformDownloadUrl)
-	terraformDownloadZipPath := fmt.Sprintf("%s/tools/terraform.zip", config.K1FolderPath)
-	err = downloadFile(terraformDownloadZipPath, terraformDownloadUrl)
-	if err != nil {
-		log.Println("error reading terraform file")
-		return err
-	}
+		unzipDirectory := fmt.Sprintf("%s/tools", config.K1FolderPath)
+		unzip(terraformDownloadZipPath, unzipDirectory)
 
-	unzipDirectory := fmt.Sprintf("%s/tools", config.K1FolderPath)
-	unzip(terraformDownloadZipPath, unzipDirectory)
+		err = os.Chmod(unzipDirectory, 0777)
+		if err != nil {
+			//return err
+		}
 
-	err = os.Chmod(unzipDirectory, 0777)
-	if err != nil {
-		return err
-	}
+		err = os.Chmod(fmt.Sprintf("%s/terraform", unzipDirectory), 0755)
+		if err != nil {
+			//return err
+		}
+		os.RemoveAll(fmt.Sprintf("%s/terraform.zip", toolsDirPath))
 
-	err = os.Chmod(fmt.Sprintf("%s/terraform", unzipDirectory), 0755)
-	if err != nil {
-		return err
-	}
-	os.RemoveAll(fmt.Sprintf("%s/terraform.zip", toolsDirPath))
-
-	helmVersion := config.HelmVersion
-	helmDownloadUrl := fmt.Sprintf(
-		"https://get.helm.sh/helm-%s-%s-%s.tar.gz",
-		helmVersion,
-		config.LocalOs,
-		config.LocalArchitecture,
-	)
-	log.Printf("Downloading terraform from %s", helmDownloadUrl)
-	helmDownloadTarGzPath := fmt.Sprintf("%s/tools/helm.tar.gz", config.K1FolderPath)
-	err = downloadFile(helmDownloadTarGzPath, helmDownloadUrl)
-	if err != nil {
-		return err
-	}
+		wg.Done()
 
-	helmTarDownload, err := os.Open(helmDownloadTarGzPath)
-	if err != nil {
-		log.Panicf("could not read helm download content")
-	}
+	}()
 
-	extractFileFromTarGz(
-		helmTarDownload,
-		fmt.Sprintf("%s-%s/helm", config.LocalOs, config.LocalArchitecture),
-		config.HelmClientPath,
-	)
-	err = os.Chmod(config.HelmClientPath, 0755)
-	if err != nil {
-		return err
-	}
+	go func() {
```

Comment:
Maybe we should use named functions here, to make tis code better to read, also cleaner. 

This function is the one most improved by been parallelized.